backend/open_webui/routers/chats.py

```diff
@@ -694,3 +762,112 @@ async def delete_all_tags_by_id(id: str, user=Depends(get_verified_user)):
         raise HTTPException(
             status_code=status.HTTP_401_UNAUTHORIZED, detail=ERROR_MESSAGES.NOT_FOUND
         )
+
+
+def extract_file_ids_from_chat(chat):
```

Comment:
Oh right... this is where they make all of the random datastructures in the middleware isn't it so there's no defined model?